mlflow/pyfunc/__init__.py

```diff
@@ -628,6 +633,35 @@ def __repr__(self):
         return yaml.safe_dump({"mlflow.pyfunc.loaded_model": info}, default_flow_style=False)
 
 
+def check_requirements_and_local_installed_mismatch(local_path):
+    req_file_path = os.path.join(local_path, "requirements.txt")
+    if not os.path.exists(req_file_path):
+        return
+    mismatch_items = []
+    for req in _parse_requirements(req_file_path, is_constraint=False):
+        req_str_splits = req.req_str.split("==")
+        if len(req_str_splits) == 2:
+            package = req_str_splits[0].strip()
+            req_version = req_str_splits[1].strip()
+            module = _convert_package_name_to_module_name(package)
+            installed_version = _get_installed_version(package, module)
+            if req_version != installed_version:
+                mismatch_items.append((req.req_str, installed_version))
+
+    if len(mismatch_items) > 0:
+        mismatch_items_str = " ,".join(
+            [
+                f"dependency {req_str} but version {installed_version} installed"
+                for req_str, installed_version in mismatch_items
+            ]
+        )
```

Comment:
Can we print out a table like this?

|Package|Expected|Installed|
|:-|:-|:-|
|scikit-learn|1.0|2.0|
|pandas|1.1|1.2|